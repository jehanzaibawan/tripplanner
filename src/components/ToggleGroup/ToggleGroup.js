import React, { useState, useEffect } from "react";
import { makeStyles } from "@material-ui/core/styles";
import ToggleButtonGroup from "@material-ui/lab/ToggleButtonGroup";
import ToggleButton from "@material-ui/lab/ToggleButton";
import Icon from "@material-ui/core/Icon";
import PropTypes from "prop-types";

const useStyles = makeStyles(theme => ({
  leftIcon: {
    marginRight: theme.spacing(1)
  }
}));

const ToggleGroup = props => {
  const [option, setOption] = useState(props.default);

  useEffect(() => {
    props.updateFilter(option);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [option]);

  const handleChange = (event, selectedValue) => {
    setOption(selectedValue);
  };

  const classes = useStyles();

  const renderChildren = () => {
    if (Array.isArray(props.options) && props.options.length > 0)
      return props.options.map((opt, index) => {
        return (
          <ToggleButton key={index + 1} value={opt.value}>
            <Icon className={classes.leftIcon}>{opt.icon}</Icon>
            <strong>{opt.value}</strong>
          </ToggleButton>
        );
      });
  };

  return (
    <ToggleButtonGroup
      size="small"
      value={option}
      exclusive
      onChange={handleChange}
    >
      {renderChildren()}
    </ToggleButtonGroup>
  );
};

ToggleGroup.propTypes = {
  options: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  default: PropTypes.string.isRequired,
  updateFilter: PropTypes.func.isRequired
};

export default ToggleGroup;
