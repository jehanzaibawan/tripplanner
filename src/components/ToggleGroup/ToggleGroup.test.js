import React from "react";
import renderer from "react-test-renderer";
import ToggleGroup from "./ToggleGroup";

it("renders a snapshot", () => {
  const props = {
    options: [
      { value: "cheapest", label: "Cheapest", icon: "low_priority" },
      { value: "fastest", label: "Fastest", icon: "fast_forward" }
    ],
    default: "cheapest",
    updateFilter: jest.fn()
  };
  const tree = renderer.create(<ToggleGroup {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
