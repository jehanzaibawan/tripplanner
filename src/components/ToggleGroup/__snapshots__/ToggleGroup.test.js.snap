// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  className="MuiToggleButtonGroup-root"
>
  <button
    className="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-sizeSmall"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
    value="cheapest"
  >
    <span
      className="MuiToggleButton-label"
    >
      <span
        aria-hidden={true}
        className="material-icons MuiIcon-root makeStyles-leftIcon-1"
      >
        low_priority
      </span>
      <strong>
        cheapest
      </strong>
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeSmall"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
    value="fastest"
  >
    <span
      className="MuiToggleButton-label"
    >
      <span
        aria-hidden={true}
        className="material-icons MuiIcon-root makeStyles-leftIcon-1"
      >
        fast_forward
      </span>
      <strong>
        fastest
      </strong>
    </span>
  </button>
</div>
`;
